### PR TITLE
Update facia architecture documentation

### PR DESCRIPTION
**Pull request by TBonnin**
## What does this change?
Updating and adding details to Fronts architecture docs.
Main new document is here: https://github.com/guardian/frontend/blob/8bfff0bf3077cbf3dfd0b85679588ecf4dcee1c9/docs/02-architecture/02-fronts-architecture.md
I have also moved the section related to monitoring and alerts into its own file.

Let me know if this is understandable. Don't hesitate to correct my english or fix typos. Please edit the file directly in github, that would be faster. Thank you @guardian/dotcom-platform 

## What is the value of this and can you measure success?
Better docs 📜 🏅

### CODE REVIEW

File: docs/02-architecture/02-fronts-architecture.md [hunk 1]
```diff
@@ -1,69 +1,110 @@
-# Fronts Architecture
+# Fronts architecture
 
 ![Fronts architecture](images/fronts-archirecture.png)
 
-The goal of the Fronts architecture is serve each front via a _single_ blocking call to a persitence layer.
+Rendering a front requires to fetch a lot of information from multiple sources (CAPI, Fronts config, Fronts collections). In order to avoid doing this grueling work inline when an user request is received, all data necessary to render a front is prefetched and aggregated in a single location (S3).
```
Comment (jfsoul) [inline docs/02-architecture/02-fronts-architecture.md]: requires ~~to fetch~~ fetching

~~grueling~~ gruelling

sorry 🤓

File: docs/02-architecture/02-fronts-architecture.md [hunk 2]
```diff
@@ -1,69 +1,110 @@
...
+# Fronts architecture
 
 ![Fronts architecture](images/fronts-archirecture.png)
 
-The goal of the Fronts architecture is serve each front via a _single_ blocking call to a persitence layer.
+Rendering a front requires to fetch a lot of information from multiple sources (CAPI, Fronts config, Fronts collections). In order to avoid doing this grueling work inline when an user request is received, all data necessary to render a front is prefetched and aggregated in a single location (S3). 
 
-The system produces static aggregation files, offline, one for each front (`pressed.json`). Each file contains all the content required for one front; it is a complete snapshot of its required ContentApi data, structured by collection, and overlaid by the temporary fronts-only article metdata. Each fulfills the single call made by the Facia app in order to service a request for that front.
+In a nutshell:
+- [Facia press](#facia-press): for each front, fetch all necessary data, aggregate it and store it into S3
```
Comment (jfsoul) [inline docs/02-architecture/02-fronts-architecture.md]: store it in~~to~~ s3

File: docs/02-architecture/02-fronts-architecture.md [hunk 3]
```diff
@@ -1,69 +1,110 @@
...
+- [Facia](#facia): read the aggregated data from S3 and render the front
 
-The global list of defined fronts - and the definitions of the collections on those fronts - is held in a single configuration file (`config.json`). This file is edited using the Configuration Tool. It is polled and referred to by each component in the architecture. 
+## Facia Press
+It is not an user facing service, it processes events published to queues.
```
Comment (jfsoul) [inline docs/02-architecture/02-fronts-architecture.md]: not a~~n~~ user facing

File: docs/02-architecture/02-fronts-architecture.md [hunk 4]
```diff
@@ -1,69 +1,110 @@
...
+There are 2 queues:
+- one queue populated by a cron job running in the Admin app.
+- one queue where the Fronts tool (fronts.gutools.co.uk) publishes events to when a front is being updated.
```
Comment (jfsoul) [inline docs/02-architecture/02-fronts-architecture.md]: events ~~to~~ when

File: docs/02-architecture/02-fronts-architecture.md [hunk 5]
```diff
@@ -1,69 +1,110 @@
...
+The cron job is very important since content for a front can be coming from CAPI and change without any update from the Fronts tool.
 
-The running order of articles in each collection - and various tweaks to those articles - are represented in multiple `collection.json` files. These are individually edited using the Fronts Tool and persisted to S3. These `collection.json` files dictate which articles need to be requested from the ContentApi during the production of each front's `pressed.json` files.
+### Pressing a front:
+Reacting to a new event published to the queues described above Facia-press then:
 
-As soon as a collection is edited, its id is added to a queue consumed the Presser, which will re-create a `pressed.json` file for any front containing that collection. Additionally, each front is periodically queued for re-pressing by the Admin app; fronts have differential rates of periodic re-pressing, to reflect their differing priorities.
+1. Get the collections that compose the front by fetching the Fronts config at `s3://facia-tool-store/STAGE/frontsapi/config/config.json` in the CMS Fronts account. This file is edited using the Configuration Tool
```
Comment (jfsoul) [inline docs/02-architecture/02-fronts-architecture.md]: link to the config tool here?

File: docs/02-architecture/02-fronts-architecture.md [hunk 6]
```diff
@@ -1,69 +1,110 @@
...
+## Facia Press
+It is not an user facing service, it processes events published to queues.
+There are 2 queues:
+- one queue populated by a cron job running in the Admin app.
+- one queue where the Fronts tool (fronts.gutools.co.uk) publishes events to when a front is being updated.
+The cron job is very important since content for a front can be coming from CAPI and change without any update from the Fronts tool.
 
-The running order of articles in each collection - and various tweaks to those articles - are represented in multiple `collection.json` files. These are individually edited using the Fronts Tool and persisted to S3. These `collection.json` files dictate which articles need to be requested from the ContentApi during the production of each front's `pressed.json` files.
+### Pressing a front:
+Reacting to a new event published to the queues described above Facia-press then:
```
Comment (jfsoul) [inline docs/02-architecture/02-fronts-architecture.md]: Maybe reword this? How about:

> In reaction to a new event published to the queues described above, Facia-press will:

File: docs/02-architecture/02-fronts-architecture.md [hunk 7]
```diff
@@ -1,69 +1,110 @@
...
-## Service Level Metrics & Alerts
+In the same file, here is an example of one collection config
+```
+"12554364-5a48-4e1a-a931-4dbcd144a12a": {
+    "displayName": "multimedia",
+    "type": "fixed/small/slow-I"
+}
+```
+Note: 
+- A collection type correspond to its layout.
```
Comment (jfsoul) [inline docs/02-architecture/02-fronts-architecture.md]: corresponds

File: docs/02-architecture/02-fronts-architecture.md [hunk 8]
```diff
@@ -1,69 +1,110 @@
...
+- Facia-press is using the [Facia scala client](https://github.com/guardian/facia-scala-client) to interact with the data store in the CMS Fronts bucket. (No direct call to S3)
 
-A large number of ContentApi requests are made in order to produce each front's `pressed.json` file. Due to this dependency, the Presser is the point in the architecture that has the most potential for failure. The metrics and alerts relate principally to its ongoing ability to produce `pressed.json` files.
+2. Get the curated stories/content for each collections of the fronts by fetching the collection data files at `s3://facia-tool-store/STAGE/frontsapi/collection/COLLECTION/ID/collection.json` in the CMS Fronts account. This file is edited using the Fronts Tool
 
-Some of the metrics are to designed to identify terminally unhealthy processes. Others are informational and indicate problems in the broader network (notably ContentApi). Others monitor the speed at which updates travel through the architecture.
+Each collection would contain the `id` and `metadata` for all their running stories that have been curated by editors using the Fronts tool.
```
Comment (jfsoul) [inline docs/02-architecture/02-fronts-architecture.md]: collection ~~would~~ will contain

File: docs/02-architecture/02-fronts-architecture.md [hunk 9]
```diff
@@ -1,69 +1,110 @@
...
+This is an additional request to CAPI to fetch SEO and other metadata for the front page
 
-* __Metrics__  : if the number of *consecutive* failures of the Presser to produce `pressed.json` files exceeds N, the healthcheck fails. The failure count is reset by any successful pressing. 
+6. Finally aggregate all the data fetched in the previous steps and store the result in its json form in S3 (`s3://aws-frontend-store/STAGE/frontsapi/pressed/live/FRONTS/fapi/pressed.json in Frontend account`)
 
-* __Consequence__ : Instance terminates.
+## Facia
 
-### Statictical press failures - CRON presser (facia-press)
+As mentioned above when Facia receive a request for a front, it:
```
Comment (jfsoul) [inline docs/02-architecture/02-fronts-architecture.md]: receives

File: docs/02-architecture/02-fronts-architecture.md [hunk 10]
```diff
@@ -1,69 +1,110 @@
...
+Each front collection would be rendered into a `container`. A container is always full-width.
 
-### Statictical press failures - MANUAL presser (facia-tool)
+The collection type (ex: `fixed/medium/slow-VI`) would decide the type and layout of the container. See [here](https://github.com/guardian/frontend/blob/master/common/app/slices/FixedContainers.scala#L79) and [here](https://github.com/guardian/frontend/blob/master/common/app/slices/Container.scala#L9)
 
-* __Detail__ : due to momentary ContentApi unavailability or increased latency somewhere in the network - pressing will on occasion fail. Above a certain frequency, these failures should be considered as indicative of a operational issue.   
+Each container is composed of `slices` (aka "rows"). Each [Slice definition](https://github.com/guardian/frontend/blob/master/common/app/slices/Slice.scala) specifies columns info and css classname.
 
-* __Metrics__  : 50 failures within 15 minutes, monitored in CloudWatch. 
+Each row would contained cards.
```
Comment (jfsoul) [inline docs/02-architecture/02-fronts-architecture.md]: Each row contains cards?

File: docs/02-architecture/02-fronts-architecture.md [hunk 11]
```diff
@@ -1,69 +1,110 @@
...
 
-* __Consequence__ : PagerDuty alarm
+A card displays the information of one story/content.
 
-### ContentAPI invalid responses
+## Glossary:
 
-* __Detail__ : the Presser (and also the fronts tools) are dependent on ContentApi results being well-formed. Occasionally ContentApi returns intermittently malformed results.
+Here is the defition of a few words and expression related to fronts:
```
Comment (jfsoul) [inline docs/02-architecture/02-fronts-architecture.md]: expressions

File: docs/02-architecture/02-fronts-architecture.md [hunk 12]
```diff
@@ -1,69 +1,110 @@
...
 
-* __Metrics__  : (a) 50 malformed results within 15 minutes, monitored in CloudWatch; (b) a single malformed result monitored by the facia-tool UI.
+_Snap_:
+A way to include arbitrary content into a collection (rather than a story). 
+A snap can be a link (often used to show an interactive on the front) or dynamic content (latest story for given tag)
 
-* __Consequence__ : (a) PagerDuty alarm; (b) a "red" alert in the UI with an option to "try again".
+_Trasher_:
```
Comment (jfsoul) [inline docs/02-architecture/02-fronts-architecture.md]: Thrasher! 😄

**Review by jfsoul — CHANGES_REQUESTED** [review]
This is really great, I especially love the glossary. 💯 

I am afraid I have just been a grammar nerd and suggested some corrections.

**Review by jfsoul — APPROVED** [review]
Made my corrections, only suggestion remaining was about the link to the config tool.  I will leave that up to you to decide on.

**Comment by TBonnin** [comment]
Thanks @jfsoul

**Review by NataliaLKB — APPROVED** [review]
Thanks for doing this! Really helpful for the team

File: docs/02-architecture/02-fronts-architecture.md [hunk 13]
```diff
@@ -1,69 +1,110 @@
...
+        "7aa73028-4e76-48d7-a44e-b62a15d36a65",
+        "uk/business/most-viewed/regular-stories",
+        "b5058768-e4a8-4e37-a61e-5f8e4df7f8fc"
+    ],
+    "webTitle": "Economy",
+    "title": "Latest financial, market & economic news and analysis | Business",
+    "description": "Latest financial, market &amp; economic news and analysis from The Guardian"
+}
+...
+```
```
Comment (kelvin-chappell) [inline docs/02-architecture/02-fronts-architecture.md]: Might be useful to distinguish between behaviour of draft and live presses?  Eg. behaviour when front is not yet published and the pressing schedule.

File: docs/02-architecture/02-fronts-architecture.md [hunk 14]
```diff
@@ -1,69 +1,110 @@
...
+The cron job is very important since content for a front can be coming from CAPI and change without any update from the Fronts tool.
 
-The running order of articles in each collection - and various tweaks to those articles - are represented in multiple `collection.json` files. These are individually edited using the Fronts Tool and persisted to S3. These `collection.json` files dictate which articles need to be requested from the ContentApi during the production of each front's `pressed.json` files.
+### Pressing a front:
+In reaction to a new event published to the queues described above, Facia-press will:
 
-As soon as a collection is edited, its id is added to a queue consumed the Presser, which will re-create a `pressed.json` file for any front containing that collection. Additionally, each front is periodically queued for re-pressing by the Admin app; fronts have differential rates of periodic re-pressing, to reflect their differing priorities.
+1. Get the collections that compose the front by fetching the Fronts config at `s3://facia-tool-store/STAGE/frontsapi/config/config.json` in the CMS Fronts account. This file is edited using the Configuration Tool
```
Comment (kelvin-chappell) [inline docs/02-architecture/02-fronts-architecture.md]: Is the Configuration tool different to the Fronts tool?  Maybe a URL here?

File: docs/02-architecture/02-fronts-architecture.md [hunk 15]
```diff
@@ -1,69 +1,110 @@
...
+Rendering a front requires fetching a lot of information from multiple sources (CAPI, Fronts config, Fronts collections). In order to avoid doing this gruelling work inline when a user request is received, all data necessary to render a front is prefetched and aggregated in a single location (S3). 
 
-The system produces static aggregation files, offline, one for each front (`pressed.json`). Each file contains all the content required for one front; it is a complete snapshot of its required ContentApi data, structured by collection, and overlaid by the temporary fronts-only article metdata. Each fulfills the single call made by the Facia app in order to service a request for that front.
+In a nutshell:
+- [Facia press](#facia-press): for each front, fetch all necessary data, aggregate it and store it in S3
+- [Facia](#facia): read the aggregated data from S3 and render the front
 
-The global list of defined fronts - and the definitions of the collections on those fronts - is held in a single configuration file (`config.json`). This file is edited using the Configuration Tool. It is polled and referred to by each component in the architecture.
```
Comment (stephanfowler) [inline docs/02-architecture/02-fronts-architecture.md]: I think this paragraph is worth keeping. The key role of `config.json`, and knowing how it gets edited, seems important to state.

File: docs/02-architecture/02-fronts-architecture.md [hunk 16]
```diff
@@ -1,69 +1,110 @@
...
+Each row contains cards.
 
-* __Consequence__ : PagerDuty alarm
+A card displays the information of one story/content.
 
-### ContentAPI invalid responses
+## Glossary:
 
-* __Detail__ : the Presser (and also the fronts tools) are dependent on ContentApi results being well-formed. Occasionally ContentApi returns intermittently malformed results.
+Here is the definition of a few words and expressions related to fronts:
```
Comment (kelvin-chappell) [inline docs/02-architecture/02-fronts-architecture.md]: > are the definitions

File: docs/02-architecture/02-fronts-architecture.md [hunk 17]
```diff
@@ -1,69 +1,110 @@
...
 
-* __Metrics__  : (a) 50 malformed results within 15 minutes, monitored in CloudWatch; (b) a single malformed result monitored by the facia-tool UI.
+_Snap_:
+A way to include arbitrary content into a collection (rather than a story).
```
Comment (kelvin-chappell) [inline docs/02-architecture/02-fronts-architecture.md]: > in a container

File: docs/02-architecture/02-fronts-architecture.md [hunk 18]
```diff
@@ -1,69 +1,110 @@
...
+Here is the definition of a few words and expressions related to fronts:
 
-* __Metrics__  : (a) 50 malformed results within 15 minutes, monitored in CloudWatch; (b) a single malformed result monitored by the facia-tool UI.
+_Snap_:
+A way to include arbitrary content into a collection (rather than a story). 
+A snap can be a link (often used to show an interactive on the front) or dynamic content (latest story for given tag).
 
-* __Consequence__ : (a) PagerDuty alarm; (b) a "red" alert in the UI with an option to "try again".
+_Thrasher_:
+A type of container which contains a single full-width element.
```
Comment (kelvin-chappell) [inline docs/02-architecture/02-fronts-architecture.md]: > that

File: docs/02-architecture/02-fronts-architecture.md [hunk 19]
```diff
@@ -1,69 +1,110 @@
...
 
-### On edit, press within 10 seconds (not incl. cache time)
+_Dynamo_:
+A type of container often used to highlight a huge story. It's better known in the codebase as the `dynamic/package` collection type.
 
-* __Detail__ : The end-to-end process from invoking the Presser to producing a `pressed.json` file - after a collection is manually edited - should be under 10 seconds. (It typically takes about 3 seconds under normal network conditions.)
+_Groups_:
+A way to further tweak the layout of a collection. Dynamic collections can have multiple groups (standard/snaps or huge/veryBig/big/standard). Depending on the group a story/content is in, the layout of the container would be slightly altered.
```
Comment (kelvin-chappell) [inline docs/02-architecture/02-fronts-architecture.md]: > will be

File: docs/02-architecture/facia/facia-monitoring.md [hunk 20]
```diff
@@ -0,0 +1,53 @@
+# Facia Level Metrics & Alerts
+
+A large number of ContentApi requests are made in order to produce each front's `pressed.json` file. Due to this dependency, the Presser is the point in the architecture that has the most potential for failure. The metrics and alerts relate principally to its ongoing ability to produce `pressed.json` files.
+
+Some of the metrics are to designed to identify terminally unhealthy processes. Others are informational and indicate problems in the broader network (notably ContentApi). Others monitor the speed at which updates travel through the architecture.
+
+## Succesive press failures - MANUAL presser (facia-tool)
```
Comment (kelvin-chappell) [inline docs/02-architecture/facia/facia-monitoring.md]: > Successive

File: docs/02-architecture/facia/facia-monitoring.md [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+
+* __Detail__ : immediately after a collection is edited within the Fronts Editor, the Presser is invoked to re-create the `pressed.json` files for any front containing that collection.
```
Comment (kelvin-chappell) [inline docs/02-architecture/facia/facia-monitoring.md]: > regenerate?

File: docs/02-architecture/facia/facia-monitoring.md [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+
+* __Metrics__  : if the number of *consecutive* failures of the Presser to produce `pressed.json` files exceeds N, the healthcheck fails. The failure count is reset by any successful pressing. 
+
+* __Consequence__ : Instance terminates.
+
+## Succesive press failures - CRON presser (facia-press)
+
+* __Detail__ : every three minutes, the Admin app puts the ids of every front on an SQS queue. These are pulled in bunches (of up to 10) every 10 seconds by Presser instances, and `pressed.json` files are re-created for each front.
```
Comment (kelvin-chappell) [inline docs/02-architecture/facia/facia-monitoring.md]: > batches?
> regenerated?

File: docs/02-architecture/facia/facia-monitoring.md [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+
+* __Metrics__  : if the number of *consecutive* failures of the Presser to produce `pressed.json` files exceeds N, the healthcheck fails. The failure count is reset by any successful pressing. 
+
+* __Consequence__ : Instance terminates.
+
+## Statictical press failures - CRON presser (facia-press)
+
+* __Detail__ : due to momentary ContentApi unavailability or increased latency somewhere in the network - pressing will on occasion fail. Above a certain frequency, these failures should be considered as indicative of a operational issue.
```
Comment (kelvin-chappell) [inline docs/02-architecture/facia/facia-monitoring.md]: > an

File: docs/02-architecture/facia/facia-monitoring.md [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+
+* __Detail__ : due to momentary ContentApi unavailability or increased latency somewhere in the network - pressing will on occasion fail. Above a certain frequency, these failures should be considered as indicative of a operational issue.   
+
+* __Metrics__  : 50 failures within 15 minutes, monitored in CloudWatch.
+
+* __Consequence__ : PagerDuty alarm
+
+## Statictical press failures - MANUAL presser (facia-tool)
+
+* __Detail__ : due to momentary ContentApi unavailability or increased latency somewhere in the network - pressing will on occasion fail. Above a certain frequency, these failures should be considered as indicative of a operational issue.
```
Comment (kelvin-chappell) [inline docs/02-architecture/facia/facia-monitoring.md]: > an

File: docs/02-architecture/facia/facia-monitoring.md [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+
+* __Consequence__ : (a) PagerDuty alarm; (b) a "red" alert in the UI with an option to "try again".
+
+### On edit, press within 10 seconds (not incl. cache time)
+
+* __Detail__ : The end-to-end process from invoking the Presser to producing a `pressed.json` file - after a collection is manually edited - should be under 10 seconds. (It typically takes about 3 seconds under normal network conditions.)
+
+* __Metric__  : the difference between the most recent timestamp of a front's various `collection.json` files, and the timestamp of that front's `pressed.json` file. This is checked 10 seconds after any edit; the result itself should be below 10 seconds.
+
+* __Consequence__ : if the metric exceeds 10 seconds, an "orange" alert appears in the tool UI with an option to "try again". The condition occurs on occasion due to momentary high latency somewhere in network, and generally does not reproduce itself on retry.
```
Comment (kelvin-chappell) [inline docs/02-architecture/facia/facia-monitoring.md]: > recur

File: docs/02-architecture/02-fronts-architecture.md [hunk 26]
```diff
@@ -1,69 +1,110 @@
...
+- [Facia press](#facia-press): for each front, fetch all necessary data, aggregate it and store it in S3
+- [Facia](#facia): read the aggregated data from S3 and render the front
 
-The global list of defined fronts - and the definitions of the collections on those fronts - is held in a single configuration file (`config.json`). This file is edited using the Configuration Tool. It is polled and referred to by each component in the architecture. 
+## Facia Press
+It is not a user facing service, it processes events published to queues.
+There are 2 queues:
+- one queue populated by a cron job running in the Admin app.
+- one queue where the Fronts tool (fronts.gutools.co.uk) publishes events when a front is being updated.
+The cron job is very important since content for a front can be coming from CAPI and change without any update from the Fronts tool.
```
Comment (kelvin-chappell) [inline docs/02-architecture/02-fronts-architecture.md]: Perhaps mention the different pressing priorities of editorial, commercial and training pages?

**Review by kelvin-chappell — APPROVED** [review]
Very clear and useful description.
Just a couple of comments (and some trivial nitpicking).

File: docs/02-architecture/02-fronts-architecture.md (hunk 15 above)
Comment (TBonnin) [inline docs/02-architecture/02-fronts-architecture.md]: This is mentioned here https://github.com/guardian/frontend/pull/16585/files#diff-19b63a5be14591333d80dbbb4586251aR21

File: docs/02-architecture/02-fronts-architecture.md (hunk 26 above)
Comment (TBonnin) [inline docs/02-architecture/02-fronts-architecture.md]: Good idea. will do

File: docs/02-architecture/02-fronts-architecture.md (hunk 14 above)
Comment (TBonnin) [inline docs/02-architecture/02-fronts-architecture.md]: No it is host on the same domain. Just adding `/config` to the Fronts tool url leads you to the Configs

File: docs/02-architecture/02-fronts-architecture.md (hunk 13 above)
Comment (TBonnin) [inline docs/02-architecture/02-fronts-architecture.md]: 👍  https://github.com/guardian/frontend/pull/16585/commits/3c89223ac1ed246dc4d7f53f64ec667d5a2b09e3

**Comment by TBonnin** [comment]
@kelvin-chappell Thank you for the review. I think I have addressed most of your feedback

File: docs/02-architecture/02-fronts-architecture.md (hunk 26 above)
Comment (TBonnin) [inline docs/02-architecture/02-fronts-architecture.md]: Done https://github.com/guardian/frontend/pull/16585/commits/90eba948e7b14172f5c0b2c53e266fa470a86f2d

**Comment by kelvin-chappell** [comment]
👍
